libs/plugin-api/src/main/java/org/elasticsearch/plugin/api/package-info.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+/**
+ * An api used by plugin developers to implement custom Elasticsearch plugins.
+ * The package convention in plugin apis is as follows:
+ * <ul>
+ *      <li> The root package is org.elasticsearch.plugin</li>
+ *      <li> Specialised API jars have their name following the root package.
+ *           i.e. org.elasticsearch.analysis
+ *      </li>
+ *      <li> Interfaces and annotations used by plugin developers are in `api` package
+ *           i.e org.elasticsearch.analysis.api or org.elasticsearch.api
```

Comment:
o.e.plugin.analysis.api

Reply:
somehow I missed those comments.. I will fix this in https://github.com/elastic/elasticsearch/pull/89772